lib/Onnxifi/base.h

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef GLOW_ONNXIFI_BASE_H
+#define GLOW_ONNXIFI_BASE_H
+
+#include "glow/ExecutionEngine/ExecutionEngine.h"
+
+#include "onnx/onnxifi.h"
+
+#include <condition_variable>
+#include <mutex>
+
+namespace glow {
+namespace onnxifi {
+
+/// BackendId associated with the Glow backend.
+class BackendId {
+public:
+  BackendId(int id) : id_(id) {}
```

Comment:
Should this be `explicit` or do we want automatic conversions?

Reply:
These ctors likely will take more params, but for now will mark them `explicit`.